### ISSUE TITLE
[BITV]: Make sure that contrasts on interactive elements between normal state and focused state have to be 3:1

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#40711]
This is accessibility issue.

For now the contrast of components between normal state and focused state is in most cases 2:1. It have to be min. 3:1.

Examples: grey(`--color-border-maxcontrast`) <-> blue (`--color-primary-element`)

![Peek 2023-09-29 1](https://github.com/nextcloud/server/assets/6078378/c5014c31-3793-480a-8b0f-cc01be627514)
![Peek 2023-09-29 16-13](https://github.com/nextcloud/server/assets/6078378/8d76aa93-e73b-497c-83df-d0d767f6f704)

@jancborchardt @nimishavijay what could we do? Would it be a solution to increase color of `--color-primary-element`?

**Comment by susnux** [comment]
Maybe related, but we also need to change some other colors for accessibility (text contrast):

https://github.com/nextcloud/server/pull/40773

**Comment by nimishavijay** [comment]
Hm, this is a really good question. `--color-primary-element` was already made darker so that it meets the contrast requirement against a white background. It would be extremely dark if we change it again to have enough contrast against the border color

> So maybe this already fixes the problem?

@susnux It actually does help, but the contrast requirement is not met by a pretty big margin so it unfortunately doesn't fix the issue fully :/ 

Ideas that are coming to mind:

- Use a thicker border

https://github.com/nextcloud/server/assets/52440189/6ba35478-3493-4fb8-833a-bdd732511373

- Show a border with a different color (similar to a button, example in the gov.uk website)

https://github.com/nextcloud/server/assets/52440189/fc477b84-165b-4093-890e-38de09b5278a

Would any of these work @JuliaKirschenheuter ?

**Comment by jancborchardt** [comment]
Using a thicker border seems nice and that is also what some others seem to do. I’d say that’s the preferred solution which looks better than the extra border – which we already use for focus feedback but is quite present.

**Comment by susnux** [comment]
I agree with Jan the thicker border looks quite good and should not be a problem as the label within the border already requires some margin on top -> no resizing of the element when the border thickened.

**Comment by marcoambrosini** [comment]
I would keep the same double outline and activate it only on keyboard navigation. So same thickness, same design strategy, and enough contrast.

<img width="598" alt="Screenshot 2023-10-26 at 14 25 27" src="https://github.com/nextcloud/server/assets/26852655/1ed5866f-d25a-4da0-88ef-63da94db4b1f">

<img width="691" alt="Screenshot 2023-10-26 at 14 24 00" src="https://github.com/nextcloud/server/assets/26852655/1e87c7b0-1ae7-4441-ba7d-79138d4376dd">

**Comment by JuliaKirschenheuter** [comment]
Hi @marcoambrosini, @nimishavijay, 

Thanks for your suggestions!
I'm not sure that thicker border would solve a problem. Michael told about contrast difference and because of it is would tend to Marco's suggestion. That should be fin!

Thank you all!

**Comment by JuliaKirschenheuter** [comment]
closing issue, done
